px4_ros2_cpp/include/px4_ros2/utils/frame_conversion.hpp

```diff
@@ -0,0 +1,180 @@
+/****************************************************************************
+ * Copyright (c) 2024 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <Eigen/Eigen>
+#include <px4_ros2/utils/geometry.hpp>
+
+namespace px4_ros2
+{
+
+/**
+ * @brief Converts attitude from NED to ENU frame.
+ * Performs reference frame change and quaternion rotation s.t. the identity quaternion points along the x-axis within its reference frame.
+ *
+ * @param q_ned Attitude quaternion in NED frame.
+ * @return Attitude quaternion in ENU frame.
+ */
+template<typename Type>
+Eigen::Quaternion<Type> attitudeNedToEnu(const Eigen::Quaternion<Type> & q_ned);
```

Comment:
Since these are templates, you need to place all implementations to the header now.